### ISSUE TITLE
[Feature]: expose the tqdm progress bar to enable logging the progress

### ISSUE THREAD

**Issue by hugolytics** [vllm-project/vllm#6154]
### 🚀 The feature, motivation and pitch

I am running vLLM on aws batch on large datasets and would like to see the progress of the jobs, this is currently not possible. 
however [tqdm-loggable](https://pypi.org/project/tqdm-loggable/) does enable this, but then I would need to swap the tqdm driver in the llm_engine, this is however not exposed.
so my request would be either to move to tqdm-loggable since it automatically switches given a interactive vs non-interactive environment. Or expose tqdm such that the user can swap it themselves.

**Comment by w013nad** [comment]
I have a PR I need to finalize that should do this.

#5774

**Comment by sethkimmel3** [comment]
Hey @hugolytics - proposing this PR to accomplish the same thing via a state callback method: https://github.com/vllm-project/vllm/pull/9780.

**Comment by hmellor** [comment]
@hugolytics is this a feature you're still interested in?

**Comment by hugolytics** [comment]
yes, i am still interested in it! sorry, i should review my notification settings, I just randomly stumbled upon this issue!
How can I help?
